src/vulkan-renderer/application.cpp

```diff
@@ -124,8 +125,9 @@ VkResult Application::load_textures() {
     std::string texture_name = "unnamed texture";
 
     for (const auto &texture_file : m_texture_files) {
-        m_textures.emplace_back(*m_vkdevice, m_vkdevice->physical_device(), m_vkdevice->allocator(), texture_file,
-                                texture_name, m_vkdevice->graphics_queue(), m_vkdevice->graphics_queue_family_index());
+        wrapper::CpuTexture cpu_texture(texture_file, texture_name);
```

Comment:
Since `cpu_texture` is only needed to construct the `GpuTexture`, maybe it would be nicer if `CpuTexture` only handled the stb image data and called `stbi_image_free` in the destructor. This would also avoid copying into a vector.

Reply:
Are you sure because currently `stbi_image_free(texture_data);` is called in the same flow as the generation.
https://github.com/inexorgame/vulkan-renderer/blob/5c962d8f4e67368b64933b4dc3f4ac9c295e15bf/src/vulkan-renderer/wrapper/cpu_texture.cpp#L51

Reply:
I have made any changes around this part yet.

Yeah, I agree though.  It does do two heap allocations which is not necessarily good.

To get around that, we would have to store an additional member variable in the class to store the error texture data if it existed? Or, we would have to allocate the error texture data itself as a `stbi_uc *`.

I am not really sure how else to do it.

Reply:
Maybe we could have a separate resource/texture loader class that, upon failure of loading a texture from a file, just returns the cached error texture?

Reply:
Although that might be too much for this PR, up to you tbh. We can always resolve this later, it's not a huge issue now.

Reply:
Yeah, that sounds good.  We can wrap up this PR and then revisit later.

Reply:
> 
> 
> Are you sure because currently `stbi_image_free(texture_data);` is called in the same flow as the generation.
> 
> https://github.com/inexorgame/vulkan-renderer/blob/5c962d8f4e67368b64933b4dc3f4ac9c295e15bf/src/vulkan-renderer/wrapper/cpu_texture.cpp#L51

You don't need to free the data in the constructor, stb says you can do that later.